### PR TITLE
gnome: Fix gtkdoc generation

### PR DESCRIPTION
**Pull request by xclaesse**
install_scripts used to replace @BUILD_ROOT@ and @SOURCE_ROOT@ but it
    was not documented and got removed in Meson 0.58.0. gnome.gtkdoc() was
    relying on that behaviour, but it has always been broken in the case the
    source or build directory contains spaces.
    
    Fix this by changing get_include_args() to substitue paths directly
    which will then get escaped correctly.
    
    Add a unit test that builds GObject documentation which is where this
    issue has been spotted.
    
    Fixes: #8744

### CODE REVIEW

File: mesonbuild/backend/backends.py [hunk 1]
```diff
@@ -1429,7 +1429,19 @@ def generate_target_install(self, d: InstallData) -> None:
...
+        result: T.List[ExecutableSerialisation] = []
+        srcdir = self.environment.get_source_dir()
+        builddir = self.environment.get_build_dir()
+        for i in self.build.install_scripts:
+            fixed_args = []
+            for a in i.cmd_args:
+                a = a.replace('@SOURCE_ROOT@', srcdir)
+                a = a.replace('@BUILD_ROOT@', builddir)
+                fixed_args.append(a)
+            es = copy.copy(i)
```
Comment (heftig) [inline mesonbuild/backend/backends.py:1441]: The `import copy` needs to be restored, as well.

Comment (xclaesse) [reply mesonbuild/backend/backends.py]: You're right. Changed my implementation now.

**Comment by xclaesse** [comment]
@heftig can you give this new path a try please?

**Comment by heftig** [comment]
Seems to work fine with glib and gnome-builder. Thanks!

**Comment by jon-turney** [comment]
I guess the test this adds supersedes #7274

**Comment by xclaesse** [comment]
@jon-turney oh, I didn't know you made a PR for that already. I think that PR can be closed now, yes.

**Pull request merged**